### PR TITLE
Add two fields for discoveryService

### PR DESCRIPTION
**Pull request by wattli**
We need to convert yaml to ProxyConfig proto, and embedding proto makes the code less readable.

### CODE REVIEW

**Comment by rkpagadala** [comment]
why is mixer and pilot different?

**Comment by wattli** [comment]
I am working on pilot now, and will handle mixer later, and i think mixer is migrating to two different ports in the meantime.

**Comment by wattli** [comment]
@rkpagadala , not sure how to reply your comment inline, please see my previous reply on top, thanks.

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -123,8 +123,14 @@ message ProxyConfig {
   // The number of worker threads to run. Default value is number of cores on the machine.
   int32 concurrency = 16;
 
-  // Address of the discovery service exposing xDS.
+  // Deprecated, use discovery_mtls_address/discovery_plain_address instead.
```
Comment (rkpagadala) [inline mesh/v1alpha1/config.proto]: i thought we wanted to use ServerAddress

I am ok with the change, please fix mixer also before .7

Comment (wattli) [reply mesh/v1alpha1/config.proto]: We need to convert yaml to this proto, and embedded proto makes the code less readable. And as mentioned above, i will update mixer after pilot work is done.

**Pull request merged**